pkg/credentialprovider/plugin/plugin.go

```diff
@@ -361,7 +361,7 @@ func (p *pluginProvider) provide(image, podNamespace, podName string, podUID typ
 	var serviceAccountToken string
 	var saAnnotations map[string]string
 	var err error
-	var serviceAccountCacheKey string
+	var serviceAccountCacheKey, podCacheKey string
 
```

Comment:
pre-existing, but getServiceAccountToken and the kubelet functions that manage / cache tokens are not aware of service account uid or the need to regenerate a token because the service account was recreated

file a separate issue for resolving this... that could involve making getServiceAccountToken optionally take the uid of the service account if known (it is known from this call site, it is not known when getting tokens for volumes), returning the service account uid in TokenRequest responses, and making the kubelet token cache smarter about invalidating if the uid is provided and doesn't match